pkg/storage/cache.go

```diff
@@ -50,48 +52,60 @@ func WithMaxCacheSize[T any](maxElements int64) InMemoryLRUCacheOpt[T] {
 
 var _ InMemoryCache[any] = (*InMemoryLRUCache[any])(nil)
 
-func NewInMemoryLRUCache[T any](opts ...InMemoryLRUCacheOpt[T]) *InMemoryLRUCache[T] {
+func NewInMemoryLRUCache[T any](opts ...InMemoryLRUCacheOpt[T]) (*InMemoryLRUCache[T], error) {
 	t := &InMemoryLRUCache[T]{
 		maxElements: defaultMaxCacheSize,
-		closeOnce:   &sync.Once{},
+		stopOnce:    &sync.Once{},
+		stopping:    &atomic.Bool{},
 	}
 
 	for _, opt := range opts {
 		opt(t)
 	}
 
-	t.ccache = ccache.New(ccache.Configure[T]().MaxSize(t.maxElements))
-	return t
+	var err error
+	t.client, err = theine.NewBuilder[string, T](t.maxElements).Build()
+	if err != nil {
+		return nil, err
+	}
+
+	return t, nil
 }
 
 func (i InMemoryLRUCache[T]) Get(key string) T {
 	var zero T
-	item := i.ccache.Get(key)
-	if item == nil {
+	item, ok := i.client.Get(key)
+	if !ok {
 		return zero
 	}
 
-	if value, expired := item.Value(), item.Expired(); !expired && !reflect.ValueOf(value).IsZero() {
-		return value
-	}
-
-	return zero
+	return item
 }
 
 func (i InMemoryLRUCache[T]) Set(key string, value T, ttl time.Duration) {
-	i.ccache.Set(key, value, ttl)
+	// negative ttl are noop
+	i.client.SetWithTTL(key, value, 1, ttl)
 }
 
 func (i InMemoryLRUCache[T]) Delete(key string) {
-	i.ccache.Delete(key)
+	i.client.Delete(key)
 }
 
 func (i InMemoryLRUCache[T]) Stop() {
-	i.closeOnce.Do(func() {
-		i.ccache.Stop()
+	swapped := i.stopping.CompareAndSwap(false, true)
+	if !swapped {
+		return
+	}
+
+	i.stopOnce.Do(func() {
+		i.client.Close()
 	})
 }
 
+func (i InMemoryLRUCache[T]) IsReady() bool {
+	return !i.stopping.Load()
+}
```

Comment:
This is anti-intuitive. When is someone expected to call `isReady` and why is it not embedded within `Set`, `Get`, etc?